src/addReducerOverProps.tsx

```diff
@@ -0,0 +1,21 @@
+import {useReducer, Reducer, useMemo} from 'react'
+import {isFunction} from 'lodash'
+
+type AddReducerOverPropsType = <TProps, TState, TAction>(
+  reducer: (props: TProps) => Reducer<TState, TAction>,
+  initialState: TState,
+) => (props: TProps) => TProps & TState & {dispatch: (action: TAction) => void}
+
+const addReducerOverProps: AddReducerOverPropsType = (
```

Comment:
As discussed with @peterstuart, only include the "static reducer" form of `useReducer()` in ad-hok via `addReducer()`

And then here if you want a dynamic-reducer-over-props, use `addReducerOverProps()`